### DISCUSSION TITLE
Dynamic asset folders: Migration questions

### DISCUSSION THREAD

**Discussion by jensolafkoch** [statamic/cms#10906] [Troubleshooting]
I love the new dynamic asset folders, but I wonder: what would be the best way to migrate lots of entries with lots of weird named files to the new option?

I believe an automated migration program would be the best option, but I only have one app where I'd need that so far ...

Probably faster would be a manual process which would involve adding duplicates of the existing asset fields, renaming both accordingly, changing the field name in all entry YAML files, and then drag all assets from the old field to the new field so that the new subfolders come into existence ... Only caveat: You cannot drag assets around yet between fields, only from outside ... Wouldn't be field-to-field-draggability a great addition to the CP anyway? :-)

It's also not possible to just select them from another folder from the field itself due to the restriction to the folder itself. And even if restricted = true would work, that wouldn't move a file to one of the new folders.

So I believe two additional features should complement the new dynamic asset folders: option to drag files between asset field, and a integrated move option to select assets elsewhere (unrestricted) which are already there and move them to the new subfolder.

And it goes further: If you change the naming of the subfolders later from slug to id or something else, existing folders should be optionally renamed so that one doesn't work with different kind of systems.

**Comment by robdekort** [comment]
I think most of these requests were out of scope for the initial feature. But you could open feature requests for any features you think are missing here: https://github.com/statamic/ideas

**Reply by jensolafkoch** [reply]
Thanks, I opened a feature request: [https://github.com/statamic/ideas/issues/1231](https://github.com/statamic/ideas/issues/1231).
